File: frontend/src/types/report.ts
export type ReportSource = 'consultant' | 'software'
export type DiffType = 'equal' | 'insert' | 'delete'

export interface DiffSegment {
  type: DiffType
  text: string
}

export interface ExtractedReport {
  source: ReportSource
  raw_text: string
  equipment_threshold: string
  diagnostic_description: string
  improvement_suggestions: string
  page_count: number
  extraction_confidence: number
  extraction_warning: string
}

export interface FieldComparison {
  field_name: string
  field_label_zh: string
  consultant_text: string
  software_text: string
  diff_segments: DiffSegment[]
  similarity_score: number
}

export interface ComparisonResult {
  id: string
  consultant_report: ExtractedReport
  software_report: ExtractedReport
  field_comparisons: FieldComparison[]
  overall_similarity: number
  created_at: string
}

export interface UploadResponse {
  session_id: string
  consultant_page_count: number
  software_page_count: number
}

File: backend/models/report.py
from __future__ import annotations
from datetime import datetime
from typing import Literal
from pydantic import BaseModel


class ExtractedReport(BaseModel):
    source: Literal["consultant", "software"]
    raw_text: str
    equipment_threshold: str = ""   # 設備閾值/健康度
    diagnostic_description: str = ""  # 診斷說明
    improvement_suggestions: str = ""  # 改善建議
    page_count: int = 0
    extraction_warning: str = ""


class DiffSegment(BaseModel):
    type: Literal["equal", "insert", "delete"]
    text: str


class FieldComparison(BaseModel):
    field_name: str
    field_label_zh: str
    consultant_text: str
    software_text: str
    diff_segments: list[DiffSegment]


class ComparisonResult(BaseModel):
    id: str
    consultant_report: ExtractedReport
    software_report: ExtractedReport
    field_comparisons: list[FieldComparison]
    created_at: datetime


class UploadResponse(BaseModel):
    session_id: str
    consultant_page_count: int
    software_page_count: int


class CompareRequest(BaseModel):
    session_id: str
